crates/core/tedge_agent/src/software_manager/actor.rs

```diff
@@ -232,6 +241,42 @@ impl SoftwareManagerActor {
         Ok(())
     }
 
+    fn detect_self_update() -> Result<(), SoftwareManagerError> {
+        info!("Checking if tedge got self updated");
+        let current_running_version = env!("CARGO_PKG_VERSION");
+        info!("Current running version: {}", current_running_version);
+
+        let executable_path = std::env::current_exe()
+            .map_err(|e| anyhow!("Failed to retrieve running executable path due to {}", e))?;
+        let agent_binary_path = executable_path.parent().unwrap().join(TEDGE_AGENT);
```

Comment:
Question. I guess `executable_path` holds the path to the `tedge-agent` executable. Why do we need to go up to parent and join `tedge-agent`? Is it the support to the old executable name with underscore `tedge_agent`?

Reply:
The `executable_path` points to the `tedge` binary and not the `tedge-agent` symlink file (The symlink is resolved). When the `tedge` binary is replaced, after the upgrade is performed, the `executable_path` becomes stale as it is still pointing to the old non-existent binary, even though it points to the same logical entity at the same path. You can't use that `PathBuf` variable anymore as it appears **deleted**. So, I'm rebuilding the path to make sure that it points to the latest binary itself.